### PR TITLE
Improve Auth/Admission documentation

### PR DESCRIPTION
**Pull request by erictune**

### CODE REVIEW

**Comment by erictune** [comment]
@kubernetes/sig-auth

File: docs/admin/accessing-the-api.md [hunk 1]
```diff
@@ -1,72 +1,149 @@
...
+and JWT Tokens (used for service accounts).
+
+Multiple authentication modules can be specified, in which case each one is tried in sequence,
+until one of them succeeds.
+
+On GCE, Client Certificates, Password, Plain Tokens, and JWT Tokens are all enabled.
+
+If the request cannot be authenticated, it is rejected with HTTP status code 401.
+Otherwise, the user is authenticated as a specific `username`, and tht user name
+is available to subsequent steps to use in their decisions.
```
Comment (liggitt) [inline docs/admin/accessing-the-api.md]: mention that group names are available from some authentication modules?

Comment (liggitt) [inline docs/admin/accessing-the-api.md]: did you want to mention impersonation?

File: docs/admin/accessing-the-api.md [hunk 2]
```diff
@@ -1,72 +1,149 @@
...
+Authentication modules include Client Certificates, Password, and Plain Tokens,
+and JWT Tokens (used for service accounts).
+
+Multiple authentication modules can be specified, in which case each one is tried in sequence,
+until one of them succeeds.
+
+On GCE, Client Certificates, Password, Plain Tokens, and JWT Tokens are all enabled.
+
+If the request cannot be authenticated, it is rejected with HTTP status code 401.
+Otherwise, the user is authenticated as a specific `username`, and tht user name
```
Comment (liggitt) [inline docs/admin/accessing-the-api.md]: typo "the"

Comment (erictune) [inline docs/admin/accessing-the-api.md]: fixed

File: docs/admin/accessing-the-api.md (hunk 1 above)
Comment (erictune) [inline docs/admin/accessing-the-api.md]: Mentioned groups.  Not ready to write about impersonation.

File: docs/admin/accessing-the-api.md [hunk 3]
```diff
@@ -1,72 +1,151 @@
...
+# Overview
 
+Users [access the API](/docs/user-guide/accessing-the-cluster) using `kubectl`,
+client libraries, or by making REST requests.  Users include human users and
+[Kubernetes Service Accounts](http://kubernetes.io/docs/user-guide/service-accounts/).
+When a request reaches the API, it goes through several stages, illustrated in the
+following diagram:
 
-## Ports and IPs Served On
+<img src="/images/docs/admin/accessing-the-api.svg">
```
Comment (deads2k) [inline docs/admin/accessing-the-api.md]: I seem to be missing this file.

File: docs/admin/accessing-the-api.md [hunk 4]
```diff
@@ -1,72 +1,151 @@
...
+it does not have a `user` object nor does it store usernames or other information about
+users in its object store.
+
+# Authorization
+
+Once the request is authenticated as coming from a specific user,
+it moves to a generic authorization step.  This is shown as step **2** in the
+diagram. 
+
+The input to the Authorization step are attributes of the REST request, including:
```
Comment (deads2k) [inline docs/admin/accessing-the-api.md]: We have separate attributes for non-resource requests: verb and url-path.

File: docs/admin/accessing-the-api.md [hunk 5]
```diff
@@ -1,72 +1,151 @@
...
+  - the username determined by the Authentication step.
```
Comment (deads2k) [inline docs/admin/accessing-the-api.md]: if the authenticator provided groups, we get those too.

File: docs/admin/accessing-the-api.md (hunk 3 above)
Comment (liggitt) [inline docs/admin/accessing-the-api.md]: not sure how image references are supposed to be specified... absolute or relative?

**Comment by erictune** [comment]
Removed RBAC docs in favor of #643

Fixed image.

**Comment by erictune** [comment]
Giving myself LGTM

**Comment by erictune** [comment]
@johndmulhausen needs docs review

File: docs/admin/accessing-the-api.md [hunk 6]
```diff
@@ -1,72 +1,151 @@
...
+# Overview
 
+Users [access the API](/docs/user-guide/accessing-the-cluster) using `kubectl`,
+client libraries, or by making REST requests.  Users include human users and
+[Kubernetes Service Accounts](http://kubernetes.io/docs/user-guide/service-accounts/).
+When a request reaches the API, it goes through several stages, illustrated in the
+following diagram:
 
-## Ports and IPs Served On
+<img src="/images/docs/admin/access-control-overview.svg">
```
Comment (johndmulhausen) [inline docs/admin/accessing-the-api.md]: img alt text goes here

File: docs/admin/accessing-the-api.md [hunk 7]
```diff
@@ -1,72 +1,151 @@
...
   1. `Localhost Port`:
 
-          - serves HTTP
+          - is intended for testing and bootstrap, and for other components of the master node
+	    (scheduler, controller-manager) to talk to the API
+          - no TLS
           - default is port 8080, change with `--insecure-port` flag.
           - defaults IP is localhost, change with `--insecure-bind-address` flag.
-          - no authentication or authorization checks in HTTP
+          - request **bypasses* authentication and authorization modules.
```
Comment (johndmulhausen) [inline docs/admin/accessing-the-api.md]: Add the extra asterisk after 'bypasses'

File: docs/admin/accessing-the-api.md [hunk 8]
```diff
@@ -1,72 +1,151 @@
...
-or understand the details.
+This document describes how access to the Kubernetes API is controlled.
 
-Most questions about accessing the cluster are covered
-in [Accessing the cluster](/docs/user-guide/accessing-the-cluster).
+# Overview
 
+Users [access the API](/docs/user-guide/accessing-the-cluster) using `kubectl`,
+client libraries, or by making REST requests.  Users include human users and
+[Kubernetes Service Accounts](http://kubernetes.io/docs/user-guide/service-accounts/).
```
Comment (johndmulhausen) [inline docs/admin/accessing-the-api.md]: URL: /docs/user-guide/service-accounts/

File: docs/admin/accessing-the-api.md [hunk 9]
```diff
@@ -1,72 +1,151 @@
...
-cluster administrators who want to customize their cluster
-or understand the details.
+This document describes how access to the Kubernetes API is controlled.
 
-Most questions about accessing the cluster are covered
-in [Accessing the cluster](/docs/user-guide/accessing-the-cluster).
+# Overview
 
+Users [access the API](/docs/user-guide/accessing-the-cluster) using `kubectl`,
+client libraries, or by making REST requests.  Users include human users and
```
Comment (johndmulhausen) [inline docs/admin/accessing-the-api.md]: User accounts can belong to both human users and Kubernetes Service Accounts.

File: docs/admin/accessing-the-api.md [hunk 10]
```diff
@@ -1,72 +1,151 @@
...
+[Kubernetes Service Accounts](http://kubernetes.io/docs/user-guide/service-accounts/).
+When a request reaches the API, it goes through several stages, illustrated in the
+following diagram:
 
-## Ports and IPs Served On
+<img src="/images/docs/admin/access-control-overview.svg">
 
-The Kubernetes API is served by the Kubernetes apiserver process.  Typically,
-there is one of these running on a single kubernetes-master node.
+# Transport Security
```
Comment (johndmulhausen) [inline docs/admin/accessing-the-api.md]: Two hashes please

File: docs/admin/accessing-the-api.md [hunk 11]
```diff
@@ -1,72 +1,151 @@
...
 
-This document describes what ports the Kubernetes apiserver
-may serve on and how to reach them.  The audience is
-cluster administrators who want to customize their cluster
-or understand the details.
+This document describes how access to the Kubernetes API is controlled.
 
-Most questions about accessing the cluster are covered
-in [Accessing the cluster](/docs/user-guide/accessing-the-cluster).
+# Overview
```
Comment (johndmulhausen) [inline docs/admin/accessing-the-api.md]: Two hashes please.

File: docs/admin/accessing-the-api.md [hunk 12]
```diff
@@ -1,72 +1,151 @@
...
+In a typical kubernetes cluster, the API served on port 443.  A TLS connection is
+established.  The API server presents a certificate.  This certificate is
+often self-signed, so `$USER/.kube/config` on the user's machine typically
+contains the root certficate for the API server's certificate, which when specified
+is used in place of the system default root certificates.  This certificate is typically
+automatically written into your `$USER/.kube/config` when you create a cluster yourself
+using `kube-up.sh`.  If the cluster has multiple users, then the creator needs to share
+the certificate with other users.
+
+# Authentication
```
Comment (johndmulhausen) [inline docs/admin/accessing-the-api.md]: Two hashes

File: docs/admin/accessing-the-api.md [hunk 13]
```diff
@@ -1,72 +1,151 @@
...
+Otherwise, the user is authenticated as a specific `username`, and the user name
+is available to subsequent steps to use in their decisions.  Some authenticators
+may also provide the group memberships of the user, while other authenticators
+do not (and expect the authorizer to determine these).
+
+While Kubernetes uses "usernames" for access control decisions and in request logging,
+it does not have a `user` object nor does it store usernames or other information about
+users in its object store.
+
+# Authorization
```
Comment (johndmulhausen) [inline docs/admin/accessing-the-api.md]: Two hashes

File: docs/admin/accessing-the-api.md [hunk 14]
```diff
@@ -1,72 +1,151 @@
...
+
+The Authorization step is designed to operate on attributes that are likely to be common to most
+REST requests, such as object name, kind, etc.  This is intended to facilitate interation with
+existing organization-wide or cloud-provider-wide access control systems (which may handle
+other APIs besides the Kubernetes API.
+
+Access controls and policies that depend on specific fields of specific Kinds of objects
+are handled by Admission Controllers.
+
+# Admission Control
```
Comment (johndmulhausen) [inline docs/admin/accessing-the-api.md]: Two hashes

File: docs/admin/accessing-the-api.md [hunk 15]
```diff
@@ -1,72 +1,151 @@
...
+In addition to rejecting objects, admission controllers can also set complex defaults for
+fields.
+
+The available Admission Control Modules are described [here](docs/admin/admission-controllers/).
+
+Once a request passes all admission controllers, it is validated using the validation routines
+for the corresponding API object, and then written to the object store (shown as step **4**).
+
+
+# API Server Ports and IPs
```
Comment (johndmulhausen) [inline docs/admin/accessing-the-api.md]: Two hashes

File: docs/admin/accessing-the-api.md [hunk 16]
```diff
@@ -1,72 +1,151 @@
...
+following diagram:
 
-## Ports and IPs Served On
+<img src="/images/docs/admin/access-control-overview.svg">
 
-The Kubernetes API is served by the Kubernetes apiserver process.  Typically,
-there is one of these running on a single kubernetes-master node.
+# Transport Security
+
+In a typical kubernetes cluster, the API served on port 443.  A TLS connection is
```
Comment (johndmulhausen) [inline docs/admin/accessing-the-api.md]: Kubernetes (caps)

**Comment by erictune** [comment]
All comments addressed.

**Comment by erictune** [comment]
No other changes.  Taking liberty of adding docs LGTM.

**Pull request merged**